Google.Api.Gax.Grpc/Rest/RestMethod.cs

```diff
@@ -32,25 +31,26 @@ internal class RestMethod
     /// </summary>
     internal string FullName { get; }
 
-    private RestMethod(MethodDescriptor protoMethod, JsonParser parser, HttpRuleTranscoder transcoder) =>
-        (_protoMethod,  _parser, FullName, _transcoder) =
-            (protoMethod, parser, $"/{protoMethod.Service.FullName}/{protoMethod.Name}", transcoder);
+    private RestMethod(ApiMetadata apiMetadata, MethodDescriptor protoMethod, JsonParser parser, HttpRuleTranscoder transcoder) =>
+        (_apiMetadata, _protoMethod,  _parser, FullName, _transcoder) =
+        (apiMetadata, protoMethod, parser, $"/{protoMethod.Service.FullName}/{protoMethod.Name}", transcoder);
 
     /// <summary>
     /// Creates a <see cref="RestMethod"/> representation from the given protobuf method representation.
     /// </summary>
+    /// <param name="apiMetadata">The metadata for the API that this method is part of.</param>
     /// <param name="method">The protobuf method to represent.</param>
     /// <param name="parser">The JSON parser to use when parsing requests.</param>
     /// <returns>A representation of the method that can be used to handle HTTP requests/responses.</returns>
-    internal static RestMethod Create(MethodDescriptor method, JsonParser parser)
+    internal static RestMethod Create(ApiMetadata apiMetadata, MethodDescriptor method, JsonParser parser)
     {
         var rule = method.GetOptions().GetExtension(AnnotationsExtensions.Http);
         if (rule is null)
         {
             throw new ArgumentException($"Method {method.Name} in service {method.Service.Name} has no HTTP rule");
         }
         var transcoder = new HttpRuleTranscoder(method.FullName, method.InputType, rule);
```

Comment:
Any reason why you just don't pass the ApiMetadata to the HttpRuleTranscoder and perform the check for int enums while transcoding. (And you can then hide TranscodingOutput :))
I think it'll be cleaner but I'm fine either way.